CONTRIBUTING.md

```diff
@@ -0,0 +1,388 @@
+process can be quite lengthy, and some amount of patience is required. If you find
+that you've been waiting for a pull request to be given attention for several
+months, there may be a number of reasons for this, some of which you can do something
+about:
+
+  - It may be because of a feature freeze due to an upcoming release. During this time,
+    only bug fixes are taken into consideration. If your pull request is a new feature,
+    it will not be prioritized until the release is over. Wait for release.
+  - It may be because the changes you are suggesting do not appeal to people. Rather than
+    nits and critique, which require effort and means they care enough to spend time on your
+    contribution, thundering silence is a good sign of widespread (mild) dislike of a given change
+    (because people don't assume *others* won't actually like the proposal). Don't take
+    that personally, though! Instead, take another critical look at what you are suggesting
+    and see if it: changes too much, is too broad, doesn't adhere to the
+    [coding.txt](doc/coding.txt), is dangerous or insecure, is messily written, etc.
+    Identify and address any of the issues you find. Then ask e.g. on IRC if someone could give
+    their opinion on the concept itself.
+  - It may be because your code is too complex for all but a few people. And those people
+    may not have realized your pull request even exists. A great way to find people who
+    are qualified and care about the code you are touching is the
+    [Git Blame feature](https://help.github.com/articles/tracing-changes-in-a-file/). Simply
+    find the person touching the code you are touching before you and see if you can find
+    them and give them a nudge. Don't be incessant about the nudging though.
+  - Finally, if all else fails, ask on Slack or elsewhere for someone to give your pull request
+    a look. If you think you've been waiting an unreasonably long amount of time (month+) for
+    no particular reason (few lines changed, etc), this is totally fine. Try to return the favor
+    when someone else is asking for feedback on their code, and universe balances out.
+
+
+Backporting
+-----------
+
+Security and bug fixes can be backported from `development` to release
+branches.
+If the backport is non-trivial, it may be appropriate to open an
+additional PR, to backport the change, only after the original PR
+has been merged.
+Otherwise, backports will be done in batches and
+the maintainers will use the proper `Needs backport (...)` labels
+when needed (the original author does not need to worry).
+
+A backport should contain the following metadata in the commit body:
+
+```
+Github-Pull: #<PR number>
+Rebased-From: <commit hash of the original commit>
+```
+
+Have a look at [an example backport PR](
+https://github.com/bitcoin/bitcoin/pull/16189).
+
+Also see the [backport.py script](
+https://github.com/bitcoin-core/bitcoin-maintainer-tools#backport).
```

Comment:
Yes I think so.

Reply:
In general, backports of security and bug fixes that are deemed critical, are merged to the hotfix branch. When an official release happens, this is also merged to the hotfix branch, so at that time, the hotfix branch and the general release (master) branch are even. If there are bug fixes/security fixes that require application against the release branch, these will typically be merged to the hotfix branch.

Reply:
These fixes should ALWAYS derive from commits in the development branch. They may have to be modified to work for the hotfix branch due to coupling with other changes in development. Special PR's that are required to do extensive backporting are discouraged. Generally if you are doing that, it means you need to do a new release.

However if it is decided to do a backport PR for fixes, that PR may be a "spur", which means when the next release is done, the hotfix branch may be force updated from the development and/or testnet branch, and the "spur" PR lost.

In general I am not supportive of back-propagation of PR's that were done against master/hotfix back into development, because this can cause problems. Bug fix backports **MUST** originate from PR's/code that was done and committted to the development branch first, so no reverse flows become necessary.

In general I will reject PR's made against master or hotfix without being made against development first.

Reply:
This is one reason our default branch is now development, rather than master, among others.

Reply:
Re-wrote it based on your comments, is it better now?

Reply:
Yep.